vue/src/mapstyle/rdwatchtiles.ts

```diff
@@ -10,16 +10,26 @@ import type {
 } from "maplibre-gl";
 
 export const buildObservationFilter = (
-  timestamp: number
-): FilterSpecification => [
-  "all",
-  ["<=", ["get", "timemin"], timestamp],
-  [
-    "any",
-    [">", ["get", "timemax"], timestamp],
-    ["==", ["get", "timemin"], ["get", "timemax"]],
-  ],
-];
+  timestamp: number,
+  filters: Record<string, string>
+): FilterSpecification => {
+  const filter: FilterSpecification = [
+    "all",
+    ["<=", ["get", "timemin"], timestamp],
+    [
+      "any",
+      [">", ["get", "timemax"], timestamp],
+      ["==", ["get", "timemin"], ["get", "timemax"]],
+    ],
+  ];
+
+  // Add any filters set in the UI
+  Object.entries(filters).forEach(([key, value]) => {
+    filter.push(["==", ["get", key], value]);
+  });
```

Comment:
This is the core of the filtering framework i attempted to set up. it dynamically generates maplibre filters based on the `filters` object in the store. Currently, this only allows filtering by equality, which will need to be refactored when we get to filtering by ranges, etc.